cmd/internal/planetscale_edge_database.go

```diff
@@ -234,47 +243,32 @@ func (p PlanetScaleEdgeDatabase) sync(ctx context.Context, tc *psdbconnect.Table
 	defer cancel()
 
 	var (
-		err    error
-		client psdbconnect.ConnectClient
+		err          error
+		vtgateClient vtgateservice.VitessClient
 	)
 
-	if p.clientFn == nil {
-		conn, err := grpcclient.Dial(ctx, ps.Host,
-			clientoptions.WithDefaultTLSConfig(),
-			clientoptions.WithCompression(true),
-			clientoptions.WithConnectionPool(1),
-			clientoptions.WithExtraCallOption(
-				auth.NewBasicAuth(ps.Username, ps.Password).CallOption(),
-			),
-		)
-		if err != nil {
-			return tc, 0, err
-		}
+	vtgateClient, conn, err := p.initializeVTGateClient(ctx, ps)
+	if err != nil {
+		return tc, 0, err
+	}
+	if conn != nil {
```

Comment:
For reviewers: will this correctly defer closing the connection?

Reply:
looks like it would.

when `err != nil` will it always be the case that `conn == nil`?

Reply:
Thank you both! I'll simplify.

Reply:
Realized we don't mock `conn` in testing, and that's likely the reason for that behavior 🤔  If alright, I'd like to handle mocking properly so we _don't_ need the nil check in a separate PR.